pyproject.toml

```
@@ -163,21 +164,30 @@ disable = [
 max-attributes = 12
 
 [tool.pytest.ini_options]
-# We need to ignore PytestUnraisableExceptionWarning because we have some
-# cleanup issues, and sometimes exceptions are raised in __del__ methods, for
-# example when trying to do async stuff and the event loop is already closed.
-# This could even be caused by the GC, some time after the test function that
-# has the issue was completed.
-# Please see this issue for more details:
-# https://github.com/frequenz-floss/frequenz-sdk-python/issues/982
-addopts = "-W=all -Werror -Wdefault::DeprecationWarning -Wdefault::PendingDeprecationWarning -Wdefault::pytest.PytestUnraisableExceptionWarning -vv"
+addopts = "-vv"
 testpaths = ["tests", "src"]
 asyncio_mode = "auto"
 asyncio_default_fixture_loop_scope = "function"
 required_plugins = ["pytest-asyncio", "pytest-mock"]
 markers = [
   "integration: integration tests (deselect with '-m \"not integration\"')",
 ]
+filterwarnings = [
+  "error",
+  "once::DeprecationWarning",
+  # We need to ignore PytestUnraisableExceptionWarning because we have some
+  # cleanup issues, and sometimes exceptions are raised in __del__ methods, for
+  # example when trying to do async stuff and the event loop is already closed.
+  # This could even be caused by the GC, some time after the test function that
+  # has the issue was completed.
+  # Please see this issue for more details:
+  # https://github.com/frequenz-floss/frequenz-sdk-python/issues/982
+  "default::pytest.PytestUnraisableExceptionWarning",
+  "once::PendingDeprecationWarning",
+  # We use a raw string (single quote) to avoid the need to escape special
+  # chars as this is a regex
+  'ignore:Protobuf gencode version .*exactly one major version older.*:UserWarning',
+]
```

Nitpick, but for extra clarity I would group the lines without comments first, and then put the lines with comments:

```suggestion
filterwarnings = [
  "error",
  "once::DeprecationWarning",
  "once::PendingDeprecationWarning",
  # We need to ignore PytestUnraisableExceptionWarning because we have some
  # cleanup issues, and sometimes exceptions are raised in __del__ methods, for
  # example when trying to do async stuff and the event loop is already closed.
  # This could even be caused by the GC, some time after the test function that
  # has the issue was completed.
  # Please see this issue for more details:
  # https://github.com/frequenz-floss/frequenz-sdk-python/issues/982
  "default::pytest.PytestUnraisableExceptionWarning",
  # We use a raw string (single quote) to avoid the need to escape special
  # chars as this is a regex
  'ignore:Protobuf gencode version .*exactly one major version older.*:UserWarning',
]
```